### PR TITLE
Common Utils Class Strings to improve

### PR DESCRIPTION
**Pull request by wenpos**
some fix and improvement for common util class 'Strings'

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/Strings.java [hunk 1]
```diff
@@ -60,7 +60,7 @@
 
     private static final String CURRENT_PATH = ".";
 
-    public static void spaceify(int spaces, String from, StringBuilder to) throws Exception {
+    public static void specify(int spaces, String from, StringBuilder to) throws Exception {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/Strings.java]: The name is correct as-is, it adds spaces.

File: core/src/main/java/org/elasticsearch/common/Strings.java [hunk 2]
```diff
@@ -858,7 +859,7 @@ private Strings() {
      */
     public static String substring(String s, int beginIndex, int endIndex) {
         if (s == null) {
-            return s;
+            return null;
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/Strings.java]: This is fine as-is.

File: core/src/main/java/org/elasticsearch/common/Strings.java [hunk 3]
```diff
@@ -914,7 +915,7 @@ public static String toString(ToXContent toXContent, boolean wrapInObject) {
      */
     public static String cleanTruncate(String s, int length) {
         if (s == null) {
-            return s;
+            return null;
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/Strings.java]: This is fine as-is.

File: core/src/main/java/org/elasticsearch/common/Strings.java [hunk 4]
```diff
@@ -462,18 +462,19 @@ public static String cleanPath(String path) {
...
-                // Points to current directory - drop it.
-            } else if (TOP_PATH.equals(element)) {
-                // Registering top path found.
-                tops++;
-            } else {
-                if (tops > 0) {
-                    // Merging path element with element corresponding to top path.
-                    tops--;
+            // Points to current directory - drop it.
+            if (!CURRENT_PATH.equals(element)) {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/Strings.java]: I find this a lot easier to read as it's currently written.

**Review by jasontedor — COMMENTED** [review]
Thanks for the PR @wenpos, but the code is either correct as-is or I prefer it how it already is.

File: core/src/main/java/org/elasticsearch/common/Strings.java (hunk 1 above)
Comment (wenpos) [inline core/src/main/java/org/elasticsearch/common/Strings.java]: i might misunderstand the usage，thanks

**Pull request closed without merging**